app/src/main/java/org/simple/clinic/teleconsultlog/drugduration/DrugDurationUpdate.kt

```diff
@@ -3,12 +3,13 @@ package org.simple.clinic.teleconsultlog.drugduration
 import com.spotify.mobius.Next
 import com.spotify.mobius.Update
 import org.simple.clinic.mobius.dispatch
+import org.simple.clinic.mobius.next
 
 class DrugDurationUpdate : Update<DrugDurationModel, DrugDurationEvent, DrugDurationEffect> {
 
   override fun update(model: DrugDurationModel, event: DrugDurationEvent): Next<DrugDurationModel, DrugDurationEffect> {
     return when (event) {
-      DurationChanged -> dispatch(HideDurationError)
+      is DurationChanged -> next(model.durationChanged(event.duration), HideDurationError)
       is DrugDurationSaveClicked -> drugDurationSaveClicked(event)
```

Comment:
In the body of `drugDurationSaveClicked`, we can use the `duration` from the model instead of reading it from the `DrugDurationSaveClicked` event now.

Reply:
Ah yes, good catch. I have added the duration to model later on & forgot to change it here as well.